server/chat/backend/agent/providers/__init__.py

```
@@ -109,39 +114,39 @@ def get_provider_for_model(
-            # Provider exists but not available (missing credentials)
-            env_var_hints = {
-                "openai": "OPENAI_API_KEY",
-                "anthropic": "ANTHROPIC_API_KEY",
-                "google": "GOOGLE_AI_API_KEY",
-            }
-            hint = env_var_hints.get(
-                detected_provider, f"{detected_provider.upper()}_API_KEY"
+        if mode not in ("direct", "auto"):
+            raise ValueError(
+                f"Invalid provider mode: {mode}. Use 'direct', 'auto', or 'openrouter'."
             )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)